pandas/tseries/plotting.py

```diff
@@ -288,23 +304,42 @@ def format_dateaxis(subplot, freq):
     default, changing the limits of the x axis will intelligently change
     the positions of the ticks.
     """
-    majlocator = TimeSeries_DateLocator(freq, dynamic_mode=True,
-                                        minor_locator=False,
-                                        plot_obj=subplot)
-    minlocator = TimeSeries_DateLocator(freq, dynamic_mode=True,
-                                        minor_locator=True,
-                                        plot_obj=subplot)
-    subplot.xaxis.set_major_locator(majlocator)
-    subplot.xaxis.set_minor_locator(minlocator)
-
-    majformatter = TimeSeries_DateFormatter(freq, dynamic_mode=True,
+
+    # handle index specific formatting
+    if isinstance(index, PeriodIndex):
+
+        majlocator = TimeSeries_DateLocator(freq, dynamic_mode=True,
                                             minor_locator=False,
                                             plot_obj=subplot)
-    minformatter = TimeSeries_DateFormatter(freq, dynamic_mode=True,
+        minlocator = TimeSeries_DateLocator(freq, dynamic_mode=True,
                                             minor_locator=True,
                                             plot_obj=subplot)
-    subplot.xaxis.set_major_formatter(majformatter)
-    subplot.xaxis.set_minor_formatter(minformatter)
+        subplot.xaxis.set_major_locator(majlocator)
+        subplot.xaxis.set_minor_locator(minlocator)
+
+        majformatter = TimeSeries_DateFormatter(freq, dynamic_mode=True,
+                                                minor_locator=False,
+                                                plot_obj=subplot)
+        minformatter = TimeSeries_DateFormatter(freq, dynamic_mode=True,
+                                                minor_locator=True,
+                                                plot_obj=subplot)
+        subplot.xaxis.set_major_formatter(majformatter)
+        subplot.xaxis.set_minor_formatter(minformatter)
+
+    elif isinstance(index, TimedeltaIndex):
+        from matplotlib import ticker
+        (vmin, vmax) = tuple(subplot.xaxis.get_view_interval())
+        n_decimals = int(np.ceil(np.log10(100 * 1e9 / (vmax - vmin))))
+        if n_decimals > 9:
+            n_decimals = 9
+        formatter = ticker.FuncFormatter(
+            lambda x, pos: timeTicks(x, pos, n_decimals))
+        subplot.xaxis.set_major_formatter(formatter)
+    else:
+        raise IOError('index type not supported')
+
+    # Note, DatetimeIndex does not use this
+    # interface. DatetimeIndex uses matplotlib.date directly
 
     # x and y coord info
     subplot.format_coord = lambda t, y: (
```

Comment:
Should this code be running when `index` is a TimedeltaIndex? I think it's controlling the stuff in the bottom right of the attached screenshot.

<img width="912" alt="screen shot 2017-01-05 at 8 30 12 am" src="https://cloud.githubusercontent.com/assets/1312546/21683959/45894118-d321-11e6-897b-7b1a1eb59842.png">